src/commands/build.rs

```diff
@@ -658,25 +651,27 @@ fn sign_images(image_name: &str, tag: Option<&str>) -> Result<()> {
             _,
             _,
             _,
-            Ok(registry_token),
+            _,
             Ok(github_event_name),
             Ok(github_ref_name),
-            Ok(github_worflow_ref),
             _,
+            Ok(cosign_private_key),
         ) if github_event_name != "pull_request"
-            && (github_ref_name == "live" || github_ref_name == "main") =>
+            && (github_ref_name == "live" || github_ref_name == "main")
+            && !cosign_private_key.is_empty()
+            && Path::new(COSIGN_PATH).exists() =>
         {
-            trace!("GITHUB_EVENT_NAME={github_event_name}, GITHUB_REF_NAME={github_ref_name}, GITHUB_WORKFLOW_REF={github_worflow_ref}");
+            trace!("GITHUB_EVENT_NAME={github_event_name}, GITHUB_REF_NAME={github_ref_name}");
 
-            env::set_var("GITHUB_TOKEN", registry_token);
+            debug!("On live branch");
 
-            debug!("On {github_ref_name} branch");
+            info!("Signing image: {image_digest}");
 
-            info!("Signing image {image_digest}");
+            trace!("cosign sign --key=env://COSIGN_PRIVATE_KEY {image_digest}");
 
-            trace!("cosign sign {image_digest}");
             if Command::new("cosign")
                 .arg("sign")
+                .arg("--key=env://COSIGN_PRIVATE_KEY")
```

Comment:
The section here looks weird in diff form, but I basically just moved the match arm up above the keyless match arm.